presto-main/src/main/java/com/facebook/presto/operator/WorkProcessor.java

```diff
@@ -58,6 +59,11 @@
      */
     T getResult();
 
+    default WorkProcessor<T> yield(BooleanSupplier yieldSignal)
```

Comment:
- +javadoc
- `yielding`? you're not instructing it to _yield now_
- why not `WorkProcessor<Optional<T>>`?

Reply:
> why not WorkProcessor<Optional<T>>?

Yielding is handled explicitly (similarly as being blocked). The interface is more explicit this way (no implicit meaning), but it is also possible to write processors that don't need to wrap every result in `Optional`.